anesthetic/utils.py

```diff
@@ -58,7 +94,7 @@ def compress_weights(w, u=None, ncompress=True):
         w = np.ones_like(u)
 
     if ncompress is True:
-        ncompress = channel_capacity(w)
+        ncompress = effective_samples(w, gamma=gamma)
     elif ncompress is False:
         return w
```

Comment:
Hmm, do you reckon there could be a way of combining `ncompress` and `gamma` into one? 

E.g.
```python
if ncompress is True or isinstance(ncompress, str):
    ncompress = effective_samples(w, gamma=ncompress)
```

Then `effective_samples` would need to map `gamma=True` to `gamma=1`. Maybe we can expand the `gamma` string options to also work for `'1'`, `'2'` etc.?
```python
elif isinstance(gamma, str):
    gamma = float(gamma)
```

Or am I overcomplicating this? 

The thing is, `gamma` (or `ncompress` in its place) should be propagated all the way to `plot.py`. However, in `plot.py` a kwarg `gamma` will be very opaque, whereas for the already existing kwarg `ncompress` it is clearer what it is for and it essentially plays the same role as `gamma` in determining the number of samples after compression...

Reply:
I think this could work, but the difference between `ncompress='2'` and `ncompress=2` is equally hard to parse. Perhaps calling it `strategy` or some synonym for compression strategy in this function, and let the real experts pass through `ncompress='0.5'` at the `plot.py` level.

I dislike `gamma`, and would prefer `beta` (since it is the same flavour as a thermodynamic beta).

Reply:
In this instance I was suggesting changing `ncompress` to `strategy` in the `compress_weights` function, but leaving it as `ncompress` higher up.

Another convention would be using negative values for ncompress instead of strings.

This would however be a substantial breaking change, in that at the moment negative values indicate equal weighting (this could swap to ncompress=True, which currently codes for compressing to the channel capacity).

Reply:
> In this instance I was suggesting changing `ncompress` to `strategy` in the `compress_weights` function, but leaving it as `ncompress` higher up.

Ah ok, so `strategy` instead of `ncompress` to be more descriptive. That works for me, I just want to avoid two potentially conflicting kwargs.

> Another convention would be using negative values for ncompress instead of strings.

What I like about the string option is that it fits to the two string options which we already have, `beta='entropy'` and `beta='kish'`.

> This would however be a substantial breaking change, in that at the moment negative values indicate equal weighting (this could swap to ncompress=True, which currently codes for compressing to the channel capacity).

`samples.compress(ncompress='entropy')` will also result in equally weighted samples, right? Only for `ncompress=-1` it is ensured that each sample point will be unique, right?

Reply:
I'm a bit unclear on the outcome of this discussion @williamjameshandley @lukashergt? 

You want `ncompress` changed to `strategy` in `compress_weights` function but `ncompress` left as `ncompress` everywhere else. But then you also want to replace `gamma` with `ncompress` and for `ncompress` to take on numerical positive values and also 'kish', 'entropy' and things like '0.5' etc?

Reply:
Yes, apologies, that was probably a bit difficult to follow. I have pushed the changes that I had in mind in commits https://github.com/handley-lab/anesthetic/pull/285/commits/1c71f772fea0e2325626c359d19549ea71f3c49f through https://github.com/handley-lab/anesthetic/pull/285/commits/c6fc942c544511cf215484804434b919b1047f86. Would be good for someone to have a look through that to make sure things make sense.

I've left the `ncompress` kwarg in `compress_weights` as is for now. I don't actually expect `compress_weights` to be used much on its own. It is mostly there to be called from `WeightedDataFrame.compress` where the docstring for `ncompress` works well, I think. But feel free to change to `strategy` or `degree` if you still think that would be better.